client/src/components/Connections/dataWrappers/ConnectionWrapper.js

```diff
@@ -5,32 +5,50 @@ import {connect} from 'react-redux'
 import {branch} from 'recompose'
 import {withDataProviders} from 'data-provider'
 import {isNil} from 'lodash'
-import type {ComponentType} from 'react'
 import {connectionDetailProvider} from '../../../dataProviders/connectionsDataProviders'
 import {receiveData} from './../../../actions/sharedActions'
+import type {ComponentType} from 'react'
+import type {EntityProps} from './EntityWrapper'
+import type {State} from '../../../state'
+import type {Dispatch} from '../../../types/reduxTypes'
+
+export type DispatchProps = {
+  // TODO dispatch should be in all the props, create Props base?
+  dispatch: Dispatch,
+}
+export type ConnectionProps = {
+  connections: Array<string>,
+}
 
 const EmptyEidsWrapper = (WrappedComponent: ComponentType<*>) => {
-  return (props) => {
+  return (props: DispatchProps & EntityProps) => {
     const eid1 = props.entity1.eids.join()
     const eid2 = props.entity2.eids.join()
     // dispatch empty connection ids list
-    props.dispatch(receiveData(['connections', 'detail'], {id: `${eid1}-${eid2}`, ids: []}, `connection-${eid1}-${eid2}`))
+    props.dispatch(
```

Comment:
It is not really clear to me, where is this `dispatch` prop comming from, but would it be possible to refactor it a bit so it received only `receiveDate` function and the place where `dispatch` is coming from would care for dispatching that `receiveData` action?

Reply:
Dispatch is comming from DispatchProvider (see https://github.com/verejnedigital/verejne.digital/blob/master/client/src/index.js). But I agree that this needs some more refactoring.